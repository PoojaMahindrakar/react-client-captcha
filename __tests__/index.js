import React from 'react';
import { mount } from 'enzyme';
import ClientCaptcha from '../src/index';

const defaultProps = {
  captchaCode: jest.fn()
};

const mockEvent = {
  stopPropagation: jest.fn(),
  target: {},
};

const wait = (time = 100) => new Promise(resolve => setTimeout(resolve, time));

describe('Captcha', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockEvent.target = {};
  });

  it('Should have a default export', () => {
    expect(typeof ClientCaptcha).toBe('function');
  });

  describe('Render', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = mount(
          <ClientCaptcha {...defaultProps} />
      );
    });

    afterAll(() => {
      wrapper.unmount();
    });

    it('Should render the captcha', () => {
      expect(wrapper.find(ClientCaptcha).length).toBe(1);
      expect(wrapper.find(ClientCaptcha)).toMatchSnapshot();
    });
  })

  describe('generateCode',() => {
    const wrapper = mount(
        <ClientCaptcha {...defaultProps} charsCount={6} chars="0123456789"/>
    );

    it('Should return a 6 chars length captcha code', () => {
      expect(wrapper.instance().generateCode()).toHaveLength(6)
    })

    it('Should return a captcha code with characters: 0123456789', () => {
      expect(wrapper.instance().generateCode()).toMatch(/[1-9]/g)
    })
  })

  describe('generateCaptcha',() => {
    const wrapper = mount(
        <ClientCaptcha {...defaultProps} />
    );

    it('Should get canvas object', () => {
      expect(wrapper.instance().canvasRef.current.getContext('2d')).toBeInstanceOf(CanvasRenderingContext2D)
    })

  })
});
