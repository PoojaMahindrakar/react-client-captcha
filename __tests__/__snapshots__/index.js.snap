// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Captcha Render Should render the captcha 1`] = `
<ClientCaptcha
  backgroundColor="#F2F2F2"
  captchaClassName=""
  captchaCode={[MockFunction]}
  chars="abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
  charsCount={4}
  containerClassName="captchaContainer"
  fontColor="#000"
  fontFamily="Arial, Tahoma"
  fontSize={22}
  height={40}
  retry={true}
  retryButtonClassName="retryButton"
  retryIcon="https://cdn.jsdelivr.net/npm/react-client-captcha/dist/retry.svg"
  retryIconSize={24}
  retryImgClassName=""
  width={100}
>
  <div
    className="captchaContainer"
  >
    <canvas
      className=""
      height={40}
      style={
        Object {
          "pointerEvents": "none",
        }
      }
      width={100}
    />
    <button
      className="retryButton"
      id="retryButton"
      onClick={[Function]}
    >
      <img
        alt="Re-new captcha"
        className=""
        height={24}
        src="https://cdn.jsdelivr.net/npm/react-client-captcha/dist/retry.svg"
        width={24}
      />
    </button>
  </div>
</ClientCaptcha>
`;
